### PR TITLE
Many fixes, enable Travis, etc

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,18 +1,4 @@
...
-packages
-
-# Or the files created by dart2js.
-*.dart.js
-*.dart.precompiled.js
-*.js_
-*.js.deps
-*.js.map
-*.sw?
-.idea/
```
Comment (alan-knight) [inline .gitignore]: You want it to commit the .idea directory?

Comment (kevmoo) [reply .gitignore]: No. I want users to put it in their `.gitignore_global` file.

Comment (kevmoo) [reply .gitignore]: See https://www.dartlang.org/guides/libraries/private-files

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
    1:00am time. This is presumably because of DST time-shifts. We may not be
    able to correct these dates, because midnight may not exist at a transition
    date, but we can cause the strict parsing to not fail for these dates.
+ * Require at least Dart 2.0.0-dev.48.
```
Comment (alan-knight) [inline CHANGELOG.md]: Why this particular version?

Comment (kevmoo) [reply CHANGELOG.md]: It has the fix for `tryParse`

File: .gitignore (hunk 1 above)
Comment (alan-knight) [reply .gitignore]: Ah, thanks.

File: CHANGELOG.md (hunk 2 above)
Comment (alan-knight) [reply CHANGELOG.md]: OK

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+language: dart
+
+dart:
+  - dev
+
+dart_task:
+  - test: -p vm
```
Comment (alan-knight) [inline .travis.yml]: There are tests that are specifically for the browser, can I also just add
test -p chrome
here?

**Comment by kevmoo** [comment]
I can follow-up with adding browser.

**Pull request closed without merging**